### PR TITLE
Revert "Initial paths shouldn't be normalized on save and restore (co…

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
Reverts atom/atom#6813

That PR introduced some bugs and we need to unblock the release of 0.202.

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
Fixes #6968.
Fixes #6969.

**Comment by maxbrunsfeld** [comment]
@mostafaeweda We can re-merge your PR, with some fixes, after Atom 0.202 is released.

File: src/window-event-handler.coffee [hunk 1]
```diff
@@ -24,13 +23,12 @@ class WindowEventHandler
...
-                atom.project.addPath(path.dirname(pathToOpen))
               else
-                atom.project.addPath(pathToOpen)
+                dirToOpen = path.dirname(pathToOpen)
+                if fs.existsSync(dirToOpen)
+                  atom.project.addPath(dirToOpen)
 
-            {protocol} = url.parse(pathToOpen)
-            unless fs.isDirectorySync(pathToOpen) or protocol?
+            unless fs.isDirectorySync(pathToOpen)
```
Comment (maxbrunsfeld) [inline src/window-event-handler.coffee]: @mostafaeweda The 2 problems we found that this PR introduced were both in this file:

1. `pathToOpen` can be `undefined`, if we're trying to open a new empty editor.
2. If `pathToOpen` is a URI, we still need to open it.

Comment (mostafaeweda) [inline src/window-event-handler.coffee]: 1- will check for the undefined and fix it
2- how about: if the uri starts with `atom://`, then it needs to be opened.
or how do I differentiate between the reload/restart case and the normal open cases?

Comment (maxbrunsfeld) [inline src/window-event-handler.coffee]: I think that we always need to open it. Other packages could define other 'opener` URIs, like `markdown-preview://view`, or whatever. In general, remote URIs could represent files, not just directories.

Comment (mostafaeweda) [inline src/window-event-handler.coffee]: agreed - addressed in: https://github.com/atom/atom/pull/6978

**Pull request merged**